test/check-application

```diff
@@ -447,20 +462,69 @@ class TestNavigator(testlib.MachineCase):
+        m.execute("echo 'some content' > /home/admin/newfile")
         b.mouse("[data-item='newfile']", "contextmenu")
         b.wait_in_text(".context-menu li:nth-child(6) button", "Create link")
         b.click(".context-menu button:contains('Create link')")
         b.wait_val("#create-link-original-wrapper input", "/home/admin/newfile")
         b.click("button.pf-m-link:contains('Cancel')")
 
+        b.mouse("[data-item='newfile']", "contextmenu")
+        b.wait_in_text(".context-menu li:nth-child(8) button", "Download file")
+        b.click(".context-menu button:contains('Download file')")
+        self.waitDownloadFile("*newfile*")
+
         # Delete button text should match item type: directory/file
         b.mouse("[data-item='newfile']", "contextmenu")
-        b.wait_in_text(".context-menu li:nth-child(8) button", "Delete file")
+        b.wait_in_text(".context-menu li:nth-child(10) button", "Delete file")
         b.click(".context-menu button:contains('Delete file')")
         b.click("button.pf-m-danger")
         b.wait_not_present("[data-item='newfile']")
 
+    def testDownloadUser(self):
+        b = self.browser
+        m = self.machine
+
+        # For group ownership test
+        m.execute("""
+            groupadd navigator-test
+            gpasswd -a admin navigator-test
+        """)
+
+        self.enter_navigator(False)
+        self.allowDownload()
+
+        # create file as root in the users homedir, should be able to read
+        m.execute("echo 'test content' > /home/admin/newfile")
+
+        b.mouse("[data-item='newfile']", "contextmenu")
+        b.wait_in_text(".context-menu li:nth-child(8) button", "Download file")
+        b.click(".context-menu button:contains('Download file')")
+        self.waitDownloadFile("*newfile*")
+
+        m.execute("chmod 600 /home/admin/newfile")
+        b.click("[data-item='newfile']")
+        b.wait_text("#description-list-owner-permissions dd", "Read and write")
+        b.wait_text("#description-list-group-permissions dd", "None")
+        b.wait_text("#description-list-other-permissions dd", "None")
+        b.mouse("[data-item='newfile']", "contextmenu")
+        b.wait_not_present(".context-menu button:contains('Download file')")
+
+        # group ownership test
+        m.execute("""
+            echo 'test content' > /home/admin/groupfile
+            chown :navigator-test /home/admin/groupfile
+            chmod 640 /home/admin/groupfile
+        """)
+
+        b.mouse("[data-item='groupfile']", "contextmenu")
+        b.wait_text("#description-list-owner-permissions dd", "Read and write")
+        b.wait_text("#description-list-group-permissions dd", "Read-only")
+        b.wait_text("#description-list-other-permissions dd", "None")
+        b.wait_in_text(".context-menu li:nth-child(8) button", "Download file")
+        b.click(".context-menu button:contains('Download file')")
+        self.waitDownloadFile("*groupfile*")
```

Comment:
Please add a test for a non-readable file which isn't caught by that heuristic. E.g. the group doesn't have read permission or (if you decide to add that to the heuristic), `chmod u-r`.